services/src/projects/project.rs

```diff
@@ -213,21 +215,116 @@ pub enum LayerType {
     Raster,
     Vector,
 }
+#[derive(Debug, PartialEq, Eq, Serialize, Deserialize, Clone)]
+#[allow(clippy::large_enum_variant)]
+pub enum Symbology {
+    Raster(RasterSymbology),
+    Vector(VectorSymbology),
+}
+
+#[derive(Debug, PartialEq, Serialize, Deserialize, Clone)]
+pub struct RasterSymbology {
+    pub opacity: f64,
+    pub colorizer: Colorizer,
+}
+
+impl Eq for RasterSymbology {}
 
 #[derive(Debug, PartialEq, Eq, Serialize, Deserialize, Clone)]
-pub enum LayerInfo {
-    Raster(RasterInfo),
-    Vector(VectorInfo),
+pub enum VectorSymbology {
+    Point(PointSymbology),
+    Line(LineSymbology),
+    Polygon(PolygonSymbology),
 }
 
 #[derive(Debug, PartialEq, Eq, Serialize, Deserialize, Clone)]
-pub struct RasterInfo {
-    pub colorizer: Colorizer,
+pub struct TextSymbology {
+    pub attribute: String,
+    pub fill_color: ColorParam,
+    pub stroke: StrokeParam,
+}
+
+#[derive(Debug, PartialEq, Eq, Serialize, Deserialize, Clone)]
+pub struct PointSymbology {
+    pub radius: NumberParam,
+    pub clustered: bool,
```

Comment:
I'm not sure if we should have clustered here, since the change of symbology changes the processing (?).
I guess clustering can be expressed as derived radius by attribute and text by attribute.

Reply:
That's a neat idea but currently I only support linear scaling of a feature attribute to the radius. You use some logarithmic transformation in the clustering algorithm right? Either we need to add the radius as an additional attribute then, or need a more sophisticated attribute to radius derivation.

Reply:
No, the algorithm outputs pixel widths directly, thus it is a linear mapping of the value.

Reply:
f1cadf4a3d888b01e3c63a66ad55fb940b508e78